how-to-deploy.md

```diff
@@ -0,0 +1,106 @@
+# xcube-geodb-openeo deployment
+
+This document is intended to show all necessary steps to deploy the geoDB-
+openEO server to the BC K8s stage cluster.
+
+There is quite a large number of actors which have to be correctly configured
+for the service to work:
+- a GitHub action inside this repository
+- a Dockerfile inside this repository
+- a Helm chart, residing within the bc-org k8s-configs repository
+- BC's internet provider IONOS
+- the bc-org k8s-namespaces repository
+- ArgoCD
+
+### GitHub action
+
+A GitHub action has been set up at `.github/workflows/workflow.yaml`
+This workflow:
+- runs the unit-level tests
+- builds a docker image using the steps
+    1) check out (of xcube-geodb-openeo)
+    2) get tag of release or branch to set it to the docker image
+    3) retrieve the version of geoDB-openEO
+    4) deployment phase (dev/stage/prod/ignore), to determine if docker image
+       actually shall be built
+    5) Build docker image
+        - name: `bcdev/xcube-geodb-openeo-server`, tag: fetched in previous
+          step 2
+- Uploads docker image to [quay.io](https://quay.io/repository/bcdev/xcube-geodb-openeo-server)
+    - uses
+        - `secrets.QUAY_REG_USERNAME`
+        - `secrets.QUAY_REG_PASSWORD`
+- can be configured with the following variables:
+    - `SKIP_UNITTESTS:` "0" or "1"; if set to "1", unit tests are skipped
+    - `FORCE_DOCKER_BUILD`: "1"; if set to "1", a docker image will be built and
+      uploaded to quay.io, regardless of the tag or release
+- The GH action does no change on the helm chart. This may be added later.
+
+### Dockerfile
+
+The Dockerfile is based on miniconda. It
+1) activates the base environment
+2) clones the xcube-GitHub repository
+3) checks out the branch that contains the server
+4) and installs it into the base environment
+5) then installs the current Python package `xcube-geodb-openeo` from sources, into the base
+   environment
+
+In its `CMD` call, it passes the file `/etc/config/config.yml` to the starting
+server.
+
+### Helm chart
+
+The helm chart for the xcube-geodb-openeo-server is located in [k8s-configs](https://github.com/bc-org/k8s-configs/tree/thomas_xxx_add-geodb-openeo/xcube-geodb-openeo/helm).
```

Comment:
Please remember: once the PR for thomas_xxx_add-geodb-openeo is merged, you need to ajust the path

Reply:
Thanks!